java-checks/src/main/java/org/sonar/java/checks/StringPrimitiveConstructorCheck.java

```diff
@@ -20,51 +20,85 @@
 package org.sonar.java.checks;
 
 import java.math.BigInteger;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
 import org.sonar.check.Rule;
-import org.sonar.java.checks.methods.AbstractMethodDetection;
+import org.sonar.java.checks.helpers.QuickFixHelper;
 import org.sonar.java.model.LiteralUtils;
+import org.sonar.java.model.expression.IdentifierTreeImpl;
+import org.sonar.java.model.expression.LiteralTreeImpl;
+import org.sonar.java.model.expression.TypeCastExpressionTreeImpl;
+import org.sonar.java.reporting.JavaQuickFix;
+import org.sonar.java.reporting.JavaTextEdit;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.semantic.MethodMatchers;
 import org.sonar.plugins.java.api.tree.ExpressionTree;
 import org.sonar.plugins.java.api.tree.LiteralTree;
 import org.sonar.plugins.java.api.tree.NewClassTree;
 import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
 
 @Rule(key = "S2129")
-public class StringPrimitiveConstructorCheck extends AbstractMethodDetection {
+public class StringPrimitiveConstructorCheck extends IssuableSubscriptionVisitor {
+
+  private static final String QUICK_FIX_MESSAGE = "Replace this \"%s\" constructor with %s";
 
   private static final String STRING = "java.lang.String";
   private static final BigInteger MIN_BIG_INTEGER_VALUE = BigInteger.valueOf(Long.MIN_VALUE);
   private static final BigInteger MAX_BIG_INTEGER_VALUE = BigInteger.valueOf(Long.MAX_VALUE);
 
+  private static final MethodMatchers EMPTY_STRING_MATCHER = MethodMatchers.create().ofTypes(STRING).constructor().addWithoutParametersMatcher().build();
+  private static final MethodMatchers BIG_INT_MATCHER = MethodMatchers.create().ofTypes("java.math.BigInteger").constructor().addParametersMatcher(STRING).build();
+
+  private static final MethodMatchers matchers = MethodMatchers.or(
+    EMPTY_STRING_MATCHER,
+    BIG_INT_MATCHER,
+    MethodMatchers.create().ofTypes(STRING).constructor().addParametersMatcher(STRING).build(),
```

Comment:
For the sack of readability, I believe it would be nice to extract this code into some static helper method within the class, for instance:
```java
private static MethodMatchers constructor(String type, String arg0) {
  return MethodMatchers.create()
    .ofTypes(type)
    .constructor()
    .addParametersMatcher(arg0)
    .build();
}
```
Which would allow you to write then:
```java
private static final MethodMatchers matchers = MethodMatchers.or(
    EMPTY_STRING_MATCHER,
    BIG_INT_MATCHER,
    constructor(STRING, STRING),
    constructor("java.lang.Byte", "byte"),
    constructor("java.lang.Character", "char"),
    ...

);
```